needy/__main__.py

```diff
@@ -10,6 +10,16 @@
 from .local_configuration import LocalConfiguration
 from .platform import available_platforms
 from .generator import available_generators
+from .caches.directorycache import DirectoryCache
+
+
+def __invoke_with_configured_needy(func, parameters):
```

Comment:
Wanted to deduplicate repeated code here, but ugh. I'm not sure of a clean way to do this using the `with` blocks.